### DISCUSSION TITLE
How to invalidate cache of nested query

### DISCUSSION THREAD

**Discussion by tedc-artium** [dotansimha/envelop#1612] [Q&A]
Hello all!

We have a query in the format of:

```
query {
  currentUser {
      id
      sessions(limit: ${limit}, filter: {isCompleted: true}) {
        snapshots {
          id
        }
      }
   }
}
```

We perform a mutation to add snapshots. In the course of this mutation, a session is marked complete. Our expectation is that the currentUser query should be invalidated by this mutation. However, we are forced to invalidate the query manually.

Is there a possible explanation for why the query is not invalidated as expected?

Thank you for your time and help! Happy to answer any questions.

**Comment by paulz** [comment]
I have the same question:

should `currentUser` cache be invalidated by inserting new `session`?

does it happen automatically or should we do it somehow in code?
